presenters/thread.rb

```diff
@@ -87,17 +105,28 @@ def get_paged_merged_responses(thread_id, responses, skip, limit, recursive=fals
       content = Comment.where(comment_thread_id: thread_id, "parent_ids" => []).
         where({:id => {"$in" => paged_response_ids}}).sort({"sk" => 1})
     end
-    {"responses" => merge_response_content(content), "response_count" => response_ids.length}
+    {"responses" => merge_response_content(content, thread_id), "response_count" => response_ids.length}
   end
 
   # Takes content output from Mongoid in a depth-first traversal order and
   # returns an array of first-level response hashes with content represented
-  # hierarchically, with a comment's list of children in the key "children".
-  def merge_response_content(content)
+  # hierarchically, with a comment's list of children in the key "children"
+  # and count of children comment in "child_count"
+  def merge_response_content(content, thread_id)
+    parent_child_maps = Comment.where(comment_thread_id: thread_id).map_reduce(@@map, @@reduce).out(inline: true)
```

Comment:
@robrap Yes, 1. is correct. We do not always get child comments but we need child_count.
I have made update for renaming attribute and replaces map_reduce with group/sum